### PR TITLE
docker: create docker file for relayer

### PR DESCRIPTION
**Pull request by tac0turtle**
created a docker file for the relayer.

Made it so that the config file can be a volume

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,24 +6,24 @@
 
 The Cosmos IBC `relayer` package contains a basic relayer implementation that is
 meant for users wishing to relay packets/data between sets of IBC enabled chains.
-In additon, it is well documented and intended as an example where anyone who is
+In addition, it is well documented and intended as an example where anyone who is
```
Comment (jackzampolin) [inline README.md:9]: Thank you for the massive spell checking fixes here 🤦‍♂️

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+FROM golang:alpine as BUILD
```
Comment (jackzampolin) [inline Dockerfile:1]: Do we want to specify go version here? I like just using `golang:alpine` personally, but see that you specify versions elsewhere.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+WORKDIR /relayer
```
Comment (jackzampolin) [inline Dockerfile:3]: not `WORKDIR $GOPATH/src/github.com/iqlusioninc/relayer`?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+# Copy the files from host
+COPY . .
+
+# Update and install needed deps prioir to installing the binary.
+RUN apk update && \
+    apk --no-cache add make=4.2.1-r2 git=2.24.3-r0 && \
```
Comment (jackzampolin) [inline Dockerfile:10]: do we need to specify `make` and `git` `rc` versions here?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    make install
+
+FROM alpine:edge
+
+ENV RELAYER /relayer
+
+RUN addgroup rlyuser && \
```
Comment (jackzampolin) [inline Dockerfile:17]: Why are we setting up a user account here? Shouldn't docker provide the isolation we want?

Comment (tac0turtle) [reply Dockerfile:17]:
```
When a Dockerfile doesn’t specify a USER, it defaults to executing the container using 
the root user. In practice, there are very few reasons why the container should have 
root privileges. Docker defaults to running containers using the root user. When that 
namespace is then mapped to the root user in the running container, it means that the 
container potentially has root access on the Docker host. Having an application on the 
container run with the root user further broadens the attack surface and enables an 
easy path to privilege escalation if the application itself is vulnerable to exploitation.
```

I follow this when dealing with docker containers that will get used in a non-testing env. I will be updating the gaia image to use a user as well. Its an added level of security, buts it's up to you if you'd like it or not

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+USER rlyuser
+
+# Define working directory
+WORKDIR $RELAYER
+
+# Copy binary from BUILD
+COPY --from=BUILD /go/bin/rly /usr/bin/rly
+
+ENTRYPOINT ["/usr/bin/rly"]
```
Comment (jackzampolin) [inline Dockerfile:28]: Like this

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+# Make config available ofr mutaitons
+VOLUME [ $RELAYER ]
```
Comment (jackzampolin) [inline Dockerfile:31]: Can we show an example of this maybe? Would be nice to show an example of mounting an existing relayer dir with the container.

File: Dockerfile (hunk 4 above)
Comment (tac0turtle) [reply Dockerfile:10]: we can go without, I was experimenting with a new tool (https://github.com/hadolint/hadolint) and it recommended it.

File: Dockerfile (hunk 3 above)
Comment (tac0turtle) [reply Dockerfile:3]: yea we can go that way. It doesn't really matter anymore where the file it, but keeping in gopath bring consistency.

File: Dockerfile (hunk 5 above)
Comment (jackzampolin) [reply Dockerfile:17]: I don't have a strong opinion here, but prefer simplicity to additional code.

**Pull request merged**